raiden/smart_contracts/ChannelManagerContract.sol

```diff
@@ -49,108 +46,127 @@ contract ChannelManagerContract {
             } else {
-                // If contract is not deployed(mostly committed suicide) only then call deleteChannel
-                deleteChannel(msg.sender, partner, channel_address, caller_index, partner_index);
+                // Delete channel if contract has self destructed
+                deleteChannel(msg.sender, partner, channel_address);
             }
         }
 
         channel = data.newChannel(msg.sender, partner, settle_timeout);
+
+        // Push channel address to array keeping track of all channels
+        all_channels.push(channel);
+        // Keep track of the index of the channel in the array
+        all_channels_index[channel] = all_channels.length - 1;
+
+        // Push channel address to the array keeping track of open channels
+        // for msg.sender
+        node_channels[msg.sender].push(channel);
+        // Keep track of the index of the channel in the array
+        node_index[msg.sender][partner] = node_channels[msg.sender].length - 1;
+
+        // Push channel address to the array keeping track of open channels
+        // for the partner
+        node_channels[partner].push(channel);
+        // Keep track of the index of the channel in the array
+        node_index[partner][msg.sender] = node_channels[partner].length - 1;
+
+        // add the two participants to mapping keeping track of all channel participants
+        channel_participants[channel] = [msg.sender, partner];
+
         ChannelNew(channel, msg.sender, partner, settle_timeout);
     }
 
-    function contractExists(address channel) returns (bool) {
+    function contractExists(address channel) private constant returns (bool) {
         return data.contractExists(channel);
     }
 
     function deleteChannel(
         address caller_address,
         address partner,
-        address channel_address,
-        uint caller_index,
-        uint partner_index)
+        address channel_address)
+        private
     {
-        data.deleteChannel(caller_address, partner, channel_address, caller_index, partner_index);
+        address[] our_channels = node_channels[caller_address];
+        address[] partner_channels = node_channels[partner];
+        uint caller_index = node_index[caller_address][partner];
+        uint partner_index = node_index[partner][caller_address];
+
+        // get the last element of the array
+        address our_last_element = our_channels[our_channels.length - 1];
+        // move the last element to the index of the element to be deleted
+        our_channels[caller_index] = our_last_element;
+        // decrease the size of the array by one
+        our_channels.length--;
+        // update the index of the moved element
+        node_index[caller_address][our_last_element] = caller_index;
+        // set the index of the deleted element to 0
+        // TODO: write test to make sure that setting it to 0 doesn't cause problems
```

Comment:
But this would never be possible due to this check https://github.com/czepluch/raiden/blob/cf3ce8087810bce76a5ebecfd081af6fe8c74300/raiden/smart_contracts/ChannelManagerContract.sol#L86